app/ui/browser-modern/sagas/page-sagas.js

```diff
@@ -190,10 +190,36 @@ function* getCertificateError({ pageId, url }) {
   yield put(PageActions.setPageState(pageId, { error, certificate }));
 }
 
-function* displayWebviewContextMenu({ e, webview }) {
+function* displayWebviewContextMenu({ e, pageId, webview }) {
   const menu = new Menu();
   const chosen = new Deferred();
 
+  menu.append(new MenuItem({
+    label: 'Back',
+    click: () => chosen.resolve(function* () {
+      yield put(PageEffects.navigatePageBack(pageId));
+    }),
+    enabled: webview.canGoBack(),
```

Comment:
Don't use `canGoBack` and `canGoForward` on webviews, we're not doing this anymore, since these won't work after session restore (even when populating them with history, because electron doesn't offer an official API for this yet, so we don't rely on its canGoBack/canGoForward methods for checking if navigation in history is possible, but we do our own checks instead on the local page history).

Use the page state selectors here: https://github.com/mozilla/tofino/blob/master/app/ui/browser-modern/selectors/pages.js#L127 specifically `getPageCanGoBack`, `getPageCanGoForward` and `getPageCanRefresh`.

To access app state in sagas, see https://yelouafi.github.io/redux-saga/docs/api/index.html#selectselector-args. See the session restore PR for examples.